### PR TITLE
Certification: Eigenvalue computation

### PR DESCRIPTION
**Pull request by shreyashankar**
Replaced eigenvalue computation with Scipy eigs.

### CODE REVIEW

File: cleverhans/experimental/certification/optimization.py [hunk 1]
```diff
@@ -313,29 +355,34 @@ def run_optimization(self):
...
         found_cert = self.run_one_step(self.current_eig_vec_val,
                                        self.params['small_eig_num_steps'],
-                                       smooth_val, penalty_val)
+                                       smooth_val, penalty_val,
+                                       learning_rate_val)
         if found_cert:
           return -1
       # Update penalty only if it looks like current objective is optimizes
-      if self.current_total_objective < -0.2:
+      if self.current_total_objective < -0.1:
```
Comment (AlexeyKurakin) [inline cleverhans/experimental/certification/optimization.py]: maybe declare 0.1 as a named constant,
see also https://en.wikipedia.org/wiki/Magic_number_(programming)

File: cleverhans/experimental/certification/dual_formulation.py [hunk 2]
```diff
@@ -4,20 +4,33 @@
...
+from scipy.sparse import csc_matrix
+from scipy.sparse.linalg import eigs
+from scipy.sparse.linalg import inv
+
 from cleverhans.experimental.certification import utils
 
+flags = tf.app.flags
+FLAGS = flags.FLAGS
+
+UPPER_BOUND = 1E5
```
[the remark this answers is not in the document]
Comment (shreyashankar) [reply cleverhans/experimental/certification/dual_formulation.py]: Removed UPPER_BOUND and replaced with np.inf

File: cleverhans/experimental/certification/dual_formulation.py [hunk 3]
```diff
@@ -250,3 +296,93 @@ def get_full_psd_matrix(self):
...
+    dual_feed_dict.update(zip(self.lambda_lu, lambda_lu_val))
+    dual_feed_dict.update({self.nu: nu_val})
+
+    old_matrix_h = self.sess.run(self.matrix_h, feed_dict=dual_feed_dict)
+    old_matrix_m = self.sess.run(self.matrix_m, feed_dict=dual_feed_dict)
+
+    min_eig_val_m, _ = eigs(old_matrix_m, k=1, which='LR',
+                            tol=1E-5, sigma=-0.1)
+
+    min_eig_val_m = np.real(min_eig_val_m) - 1E-5
```
[the remark this answers is not in the document]
Comment (shreyashankar) [reply cleverhans/experimental/certification/dual_formulation.py]: commented

**Comment by AlexeyKurakin** [comment]
overall looks good, but please check test error, also please re-run certify.py to make sure that it works on some data

**Pull request merged**